cache/cache.go

```diff
@@ -0,0 +1,43 @@
+package cache
+
+import (
+	"main/repo"
+	"sync"
+)
+
+type CacheInterface interface {
+	Set(value *repo.Record)
+	Get(key string) *repo.Record
+	Delete(key string)
+}
+
+type InMemoCache struct {
+	cache map[string]*repo.Record
```

Comment:
I'd suggest rewriting cache map type to `map[string][]byte` to make the cache more flexible and independent from specific implementation